main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -0,0 +1,1077 @@
+        expectedEff <- Validation.traverse(expectedEff0)(ascribeType(_, KindMatch.subKindOf(Kind.Bool), kenv, root))
+      } yield KindedAst.Expression.Ascribe(exp, expectedType.headOption, expectedEff.headOption, Type.freshVar(Kind.Star), loc)
+    case ResolvedAst.Expression.Cast(exp0, declaredType0, declaredEff0, loc) =>
+      for {
+        exp <- ascribeExpression(exp0, kenv, root)
+        declaredType <- Validation.traverse(declaredType0)(ascribeType(_, KindMatch.subKindOf(Kind.Star), kenv, root))
+        declaredEff <- Validation.traverse(declaredEff0)(ascribeType(_, KindMatch.subKindOf(Kind.Bool), kenv, root))
+      } yield KindedAst.Expression.Cast(exp, declaredType.headOption, declaredEff.headOption, Type.freshVar(Kind.Star), loc)
+    case ResolvedAst.Expression.TryCatch(exp0, rules0, loc) =>
+      for {
+        exp <- ascribeExpression(exp0, kenv, root)
+        rules <- Validation.traverse(rules0)(ascribeCatchRule(_, kenv, root))
+      } yield KindedAst.Expression.TryCatch(exp, rules, loc)
+    case ResolvedAst.Expression.InvokeConstructor(constructor, args0, loc) =>
+      for {
+        args <- Validation.traverse(args0)(ascribeExpression(_, kenv, root))
+      } yield KindedAst.Expression.InvokeConstructor(constructor, args, loc)
+    case ResolvedAst.Expression.InvokeMethod(method, exp0, args0, loc) =>
+      for {
+        exp <- ascribeExpression(exp0, kenv, root)
+        args <- Validation.traverse(args0)(ascribeExpression(_, kenv, root))
+      } yield KindedAst.Expression.InvokeMethod(method, exp, args, loc)
+    case ResolvedAst.Expression.InvokeStaticMethod(method, args0, loc) =>
+      for {
+        args <- Validation.traverse(args0)(ascribeExpression(_, kenv, root))
+      } yield KindedAst.Expression.InvokeStaticMethod(method, args, loc)
+    case ResolvedAst.Expression.GetField(field, exp0, loc) =>
+      for {
+        exp <- ascribeExpression(exp0, kenv, root)
+      } yield KindedAst.Expression.GetField(field, exp, loc)
+    case ResolvedAst.Expression.PutField(field, exp10, exp20, loc) =>
+      for {
+        exp1 <- ascribeExpression(exp10, kenv, root)
+        exp2 <- ascribeExpression(exp20, kenv, root)
+      } yield KindedAst.Expression.PutField(field, exp1, exp2, loc)
+    case ResolvedAst.Expression.GetStaticField(field, loc) => KindedAst.Expression.GetStaticField(field, loc).toSuccess
+    case ResolvedAst.Expression.PutStaticField(field, exp0, loc) =>
+      for {
+        exp <- ascribeExpression(exp0, kenv, root)
+      } yield KindedAst.Expression.PutStaticField(field, exp, loc)
+    case ResolvedAst.Expression.NewChannel(exp0, tpe0, loc) =>
+      for {
+        exp <- ascribeExpression(exp0, kenv, root)
+        tpe <- ascribeType(tpe0, KindMatch.subKindOf(Kind.Star), kenv, root)
+      } yield KindedAst.Expression.NewChannel(exp, tpe, loc)
+    case ResolvedAst.Expression.GetChannel(exp0, loc) =>
```

Comment:
Could you add a line break between every case. I think we usually do that to have some visual separation. Makes it easier to find a specific case.